### PR TITLE
Don't merge: trying to simplify #5697

### PR DESCRIPTION
**Pull request by elsirion**

### CODE REVIEW

File: fedimint-client/src/meta.rs [hunk 1]
```diff
@@ -227,6 +227,7 @@ impl<S: MetaSource + ?Sized> MetaService<S> {
 #[non_exhaustive]
 pub struct LegacyMetaSource {
     reqwest: reqwest::Client,
+    extra_override_keys: Vec<String>,
```
Comment (dpc) [inline fedimint-client/src/meta.rs:230]: Unrelated, slipped in, I assume? @elsirion

File: modules/fedimint-wallet-client/src/lib.rs [hunk 2]
```diff
@@ -1161,7 +1161,7 @@ mod tests {
             }
         );
         assert_eq!(
-            recover_scan_idxes_for_activity(TweakIdx(10), &BTreeSet::new(), |cur_idx| async move {
+            recover_scan_idxes_for_activity(TweakIdx(9), &BTreeSet::new(), |cur_idx| async move {
```
Comment (dpc) [inline modules/fedimint-wallet-client/src/lib.rs:1164]: The fact that the argument passed to this function, which acts somewhat like a start of a range is not inclusive is very weird, IMO. :D

File: fedimint-client/src/meta.rs (hunk 1 above)
Comment (elsirion) [reply fedimint-client/src/meta.rs:230]: Oh yeah :see_no_evil:

File: modules/fedimint-wallet-client/src/lib.rs (hunk 2 above)
Comment (elsirion) [reply modules/fedimint-wallet-client/src/lib.rs:1164]: Isn't it inclusive? Maybe I missed something, I just mechanically moved the `.prev()` call out of the `recover_scan_idxes_for_activity` fn into the callers.

**Comment by dpc** [comment]
Accomodated these changes in the original PR.

**Pull request closed without merging**